mastodoncustomemojis/mastodoncustomemojis.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * Name: Mastodon Custom Emojis
+ * Description: Replace emojis shortcodes in Mastodon posts with their originating server custom emojis images.
+ * Version: 1.0
+ * Author: Hypolite Petovan
+ */
+
+use Friendica\Core\Addon;
+
+function mastodoncustomemojis_install()
+{
+	Addon::registerHook('put_item_in_cache',  __FILE__, 'mastodoncustomemojis_put_item_in_cache');
+	Addon::registerHook('network_mod_init',   __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::registerHook('display_mod_init',   __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::registerHook('search_mod_init',    __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::registerHook('community_mod_init', __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::registerHook('contacts_mod_init',  __FILE__, 'mastodoncustomemojis_css_hook');
+}
+
+function mastodoncustomemojis_uninstall()
+{
+	Addon::unregisterHook('put_item_in_cache',  __FILE__, 'mastodoncustomemojis_put_item_in_cache');
+	Addon::unregisterHook('network_mod_init',   __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::unregisterHook('display_mod_init',   __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::unregisterHook('search_mod_init',    __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::unregisterHook('community_mod_init', __FILE__, 'mastodoncustomemojis_css_hook');
+	Addon::unregisterHook('contacts_mod_init',  __FILE__, 'mastodoncustomemojis_css_hook');
+}
+
+function mastodoncustomemojis_css_hook(Friendica\App $a)
+{
+	$a->page['htmlhead'] .= <<<HTML
+<!-- Style added by mastodoncustomemojis -->
+<style type="text/css">
+	.emoji.mastodon {
+		height: 1.2em;
+		vertical-align: middle;
+	}
+</style>
+
+
+HTML;
+}
+
+function mastodoncustomemojis_put_item_in_cache(Friendica\App $a, &$hook_data)
+{
+	// Mastodon uses OStatus, skipping other network protocols
+	if ($hook_data['item']['network'] != Friendica\Core\Protocol::OSTATUS) {
+		return;
+	}
+
+	$emojis = mastodoncustomemojis_get_custom_emojis_for_author($hook_data['item']['author-link']);
```

Comment:
You should use longer function names. These short ones aren't really readable 😁

Reply:
Ideally, this file would be a class, and we wouldn't need to write such long function names with this prefix!

Reply:
Yeah, we really need a solution for the addons. But at first we should get rid of the functions in the core.

Reply:
Agreed!